plugins/inputs/system/memory.go

```diff
@@ -40,6 +50,39 @@ func (s *MemStats) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
+func (s *MemStats) gatherLinuxMeminfo(acc telegraf.Accumulator) error {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Printf("Recover: %s\n", r)
+		}
+	}()
+	f, err := os.Open(linuxMeminfoPath)
+	if err != nil {
+		return fmt.Errorf("unable to open /proc/meminfo: %s", err)
+	}
+	defer f.Close()
+
+	fields := map[string]interface{}{}
+	scnr := bufio.NewScanner(f)
+	for scnr.Scan() {
+		spl := strings.Split(scnr.Text(), ":")
+
+		k := strings.Replace(strings.Replace(spl[0], ")", "", 1), "(", "_", 1)
+
+		v, err := strconv.ParseUint(strings.Trim(spl[1], " kB"), 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing value from '%s': %s", spl[1], err)
+		}
+		fields[k] = v
+	}
+	if err := scnr.Err(); err != nil {
+		return fmt.Errorf("error reading /proc/meminfo: %s", err)
+	}
+	acc.AddGauge("mem", fields, nil)
```

Comment:
I was a little confused on this. I noticed that the `system` plugin is using `AddGauge()`/`AddCounter()`  instead of `AddFields()`. However the issue which documents their use case (#1683) says to reference https://prometheus.io/docs/concepts/metric_types/ for their definition. And that definition says counter should never decrease. Yet the current code uses `AddCounter` for the current memory metric, and those values will go up and down all the time ([line 48](https://github.com/influxdata/telegraf/pull/2342/files#diff-96ce4add4e6fa11bb54396e9ae99e74eR48) up above).

So while the current code uses `AddCounter()`, I used `AddGauge()` as it reflects how the documentation says it should be used.

Perhaps we could get some clarification on the [godoc](https://godoc.org/github.com/influxdata/telegraf) for these?